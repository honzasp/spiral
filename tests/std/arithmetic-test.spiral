(program
  (import std)
  (println "literals")
  (println 42)
  (println 0.5)

  (println "\nints")
  (println (+ 7 5))
  (println (- 15 3))
  (println (* 3 4))
  (println (div 25 2))
  (println (mod 52 20))
  (println (neg 30))

  (println "\ndoubles")
  (println (+ 0.5 0.25))
  (println (- 0.75 0.25))
  (println (* 0.5 0.4))
  (println (/ 0.75 0.5))
  (println (/ 10 4))

  (println "\nmixed")
  (println (+ 0.2 20))
  (println (- 10 0.1))
  (println (* 12 0.5))
  (println (/ 3.4 2))

  (println "\nother")
  (println (neg 30))
  (println (floor 2.45))
  (println (ceil 6.2))
  (println (abs (neg 2.34)))
  (println (abs 32)))
