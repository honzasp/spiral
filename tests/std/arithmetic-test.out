literals
42
0.5

ints
12
12
12
12
12
-30

doubles
0.75
0.5
0.2
1.5
2.5

mixed
20.2
9.9
6
1.7

other
-30
2
7
2.34
32
